modules/tracker/models/pdo/SubmissionModel.php

```diff
@@ -150,4 +152,76 @@ public function getOrCreateSubmission($producerDao, $uuid)
 
         return $submissionDao;
     }
+
+    /**
+     * Get submissions associated with a given producer.
+     *
+     * @param Tracker_ProducerDao $producerDao producer DAO
+     * @param false | string $date the date in which to check
+     * @param string $branch the branch of the submission for which to search
+     * @param bool $onlyOneDay whether to only get the last day.
+     * @return Tracker_SubmissionDao submission | false
+     */
+    public function getLatestSubmissionByProducerDateAndBranch($producerDao, $date = false, $branch = 'master',
```

Comment:
@cpatrick 

This function looks good, but the documentation could be improved.

As I read it, the function will return at most one submission, the submission should be prior to the date asked for, on a specific branch, and either looking all the way back in time or only 24 hours.

I think docs are clearer as

`Get submissions associated with a given producer.` => `Get the single latest submission associated with a given producer.`

`the date in which to check` => `the returned submission must be on or earlier than this date`    and  I don't love my suggestion, especially since this takes a datetime, and so if someone gives midnight on a day it will not actually return any submissions from that day.

`the branch of the submission for which to search`  => `the branch of the submission`, what you have is fine, but I had to read a couple times to convince myself of that, so I think this is simpler

`whether to only get the last day.` => `whether to only look in the 24 hours before $date`  or `whether to only get submissions on $date`, but I think the function would have to be revised a bit to ensure that is correct, related to the ability to pass in a datetime rather than just a date.